sway-core/tests/sway_to_ir/impl_self_reassignment.ir

```diff
@@ -0,0 +1,33 @@
+script {
+    fn main() -> () {
+        local ptr { u64 } a
+
+        entry:
+        v0 = const { u64 } { u64 undef }, !1
+        v1 = const u64 0, !2
+        v2 = insert_value v0, { u64 }, v1, 0, !1
```

Comment:
@otrho this looks a bit weird to even though it's functional. The fact that we're inserting into a `const`. Is this expected?

Reply:
Yep, conceptually I was thinking functional style, so `insert_value` isn't modifying the original, but creating a new value.  `v2` is not a modified `v0`, it's a whole new value.  But this will go away with the struct refactor and everything goes on the stack.

Reply:
Right, yeah that makes sense. This is SSA anyways.. I don't know what I was thinking 😂